prov/verbs/src/verbs_eq.c

```diff
@@ -806,21 +806,52 @@ void fi_ibv_eq_remove_events(struct fi_ibv_eq *eq, struct fid *fid)
 	}
 }
 
-ssize_t fi_ibv_eq_write_event(struct fi_ibv_eq *eq, uint32_t event,
-			      const void *buf, size_t len)
+static struct fi_ibv_eq_entry  *
+fi_ibv_eq_alloc_entry(struct fi_ibv_eq *eq, uint32_t event,
+		     const void *buf, size_t len)
 {
 	struct fi_ibv_eq_entry *entry;
 
 	entry = calloc(1, sizeof(struct fi_ibv_eq_entry) + len);
 	if (!entry) {
 		VERBS_WARN(FI_LOG_EP_CTRL, "Unable to allocate EQ entry\n");
-		return -FI_ENOMEM;
+		return NULL;
 	}
 
 	entry->event = event;
 	entry->len = len;
 	memcpy(entry->entry, buf, len);
 
+	return entry;
+}
+
+/* Caller must hold eq:lock */
+ssize_t _fi_ibv_eq_write_event(struct fi_ibv_eq *eq, uint32_t event,
+			       const void *buf, size_t len)
+{
+	struct fi_ibv_eq_entry *entry;
+
+	entry = fi_ibv_eq_alloc_entry(eq, event, buf, len);
+	if (!entry) {
+		VERBS_WARN(FI_LOG_EP_CTRL, "Unable to allocate EQ entry\n");
+		return -FI_ENOMEM;
+	}
+	dlistfd_insert_tail(&entry->item, &eq->list_head);
+
+	return len;
+}
```

Comment:
This function is only called from one place, which doesn't look at the return code.  So, this function is basically:
```
entry = alloc_entry(...)
if (entry)
    dlist_insert_tail(...)
```
I would just write those 3 lines directly in the calling place.  This avoids having two functions with nearly identical signatures and names (differ only be an underscore).

Reply:
Sounds good, done.